examples/showcase/tracked-controls/js/components/vive-controls.js

```diff
@@ -0,0 +1,117 @@
+/* global AFRAME */
+
+/**
+ * Interfaces with vive controls and maps Gamepad events to
+ * vive controller buttons: trackpad, trigger, grip, menu and system
+ * It loads a vive controller model and highlight the pressed button
+ *
+ */
+AFRAME.registerComponent('vive-controls', {
+  dependencies: ['tracked-controls'],
+
+  schema: {
+    hand: {default: 'left'},
+    model: {default: true}
+  },
+
+  // buttonId
+  // 0 - trackpad
+  // 1 - trigger ( intensity value from 0.5 to 1 )
+  // 2 - grip
+  // 3 - menu ( dispatch but better for menu options )
+  // 4 - system ( never dispatched on this layer )
+  mapping: {
+    axis0: 'trackpad',
+    axis1: 'trackpad',
+    button0: 'trackpad',
+    button1: 'trigger',
+    button2: 'grip',
+    button3: 'menu',
+    button4: 'system'
+  },
+
+  init: function () {
+    var self = this;
+    this.animationActive = 'pointing';
+    this.onButtonChanged = this.onButtonChanged.bind(this);
+    this.onButtonDown = function (evt) { self.onButtonEvent(evt.detail.id, 'down'); };
+    this.onButtonUp = function (evt) { self.onButtonEvent(evt.detail.id, 'up'); };
+    this.onModelLoaded = this.onModelLoaded.bind(this);
+  },
+
+  play: function () {
+    var el = this.el;
+    el.addEventListener('buttonchanged', this.onButtonChanged);
+    el.addEventListener('buttondown', this.onButtonDown);
+    el.addEventListener('buttonup', this.onButtonUp);
+    el.addEventListener('axisMove', this.onButtonUp);
+    el.addEventListener('model-loaded', this.onModelLoaded);
+  },
+
+  pause: function () {
+    var el = this.el;
+    el.removeEventListener('buttonchanged', this.onButtonChanged);
+    el.removeEventListener('buttondown', this.onButtonDown);
+    el.removeEventListener('buttonup', this.onButtonUp);
+    el.removeEventListener('axisMove', this.onButtonUp);
+    el.removeEventLister('model-loaded', this.onModelLoaded);
+  },
+
+  update: function () {
+    var el = this.el;
+    var data = this.data;
+    var objUrl = 'url(https://media.aframe.io/controllers/vive/vr_controller_vive.obj)';
```

Comment:
and make these constants too.

Would also be good to make a util to build the URL in case we change the domain, then we only have to change in once place